packages/react-switch/src/components/Switch/useSwitchStyles.ts

```diff
@@ -1,297 +1,231 @@
-import { shorthands, makeStyles, mergeClasses } from '@griffel/react';
 import { createFocusOutlineStyle } from '@fluentui/react-tabster';
 import { tokens } from '@fluentui/react-theme';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
 import type { SwitchState } from './Switch.types';
 
 export const switchClassName = 'fui-Switch';
-const trackClassName = `${switchClassName}-track`;
-const thumbClassName = `${switchClassName}-thumb`;
+const switchThumbClassName = 'fui-Switch__thumb';
+const switchTrackClassName = 'fui-Switch__track';
 
-/**
- * Styles for the root slot
- */
 const useRootStyles = makeStyles({
-  root: {
-    '--switch-width': '40px',
-    '--switch-height': '20px',
-    '--switch-thumb-size': '14px',
-    '--switch-checked-opacity': '0',
-    '--switch-unchecked-opacity': '0',
-
+  base: {
+    boxSizing: 'border-box',
+    cursor: 'pointer',
+    height: '36px',
+    ...shorthands.padding('8px'),
     position: 'relative',
-    width: 'var(--switch-width)',
-    height: 'var(--switch-height)',
-    display: 'inline-block',
-    userSelect: 'none',
-    touchAction: 'none',
-    verticalAlign: 'bottom',
+    width: '56px',
   },
 
-  unchecked: {
-    [`:hover .${thumbClassName}`]: {
-      ':before': {
-        backgroundColor: tokens.colorNeutralStrokeAccessibleHover,
-      },
+  focusIndicator: createFocusOutlineStyle({ style: {}, selector: 'focus-within' }),
+
+  checked: {
```

Comment:
Is it possible to use pseudo classes for styling [like in `Checkbox`](https://github.com/microsoft/fluentui/pull/21837)?

Reply:
We could, I'd maybe explore it in a separate PR to evaluate perf implications of each method more thoroughly.

Reply:
This has been changed.

Reply:
After the tech sync today decided to use pseudo classes after all. I've updated the PR to reflect this.